feat(backend-common): exposed onAuth callback (#20938)
backstage/backstage: .changeset/tiny-cobras-look.md

---
'@backstage/backend-common': minor
---

> Expose an ``onAuth` handler for `git` actions to provide custom credentials

Reviewer: Expose an `onAuth` handler for `git` actions to provide custom credentials

Author: Removed the extra backtick

> '@backstage/backend-common': minor
> ---
> 
> Expose an ``onAuth` handler for `git` actions to provide custom credentials

Reviewer 2: ```suggestion
'@backstage/backend-common': patch
---

Expose an `onAuth` handler for `git` actions to provide custom credentials
```

Fix from above + switch to patch

Reviewer 2: Ah alright, don't have permission to apply this change. You'll need to apply it yourself

Author: Changed to `patch` 👍

> Expose the ``onAuth` callback so that plugins can use it to provide credentials based on the URL of the git repository.

Reviewer: ```suggestion
Expose  an `onAuth` handler for `git` actions to provide custom credentials
```

Author: Fixed 👍
